tests/python/pants_test/test_base.py

```diff
@@ -296,12 +296,18 @@ def scan():
     return set(scan())
 
   def _reset_engine(self):
-    if self._scheduler is not None:
+    if self._build_graph is not None:
       self._build_graph.reset()
+    if self._scheduler is not None:
       # Eagerly free file handles, threads, connections, etc, held by the scheduler. In theory,
       # dropping the scheduler is equivalent, but it's easy for references to the scheduler to leak.
       self._scheduler.pre_fork()
 
+  def reset_scheduler(self):
```

Comment:
I don't think we should do this in order to accommodate `AddressMapperTest` and other tests that are intentionally testing at a lower level.